server/models/post.model.js

```diff
@@ -152,56 +163,67 @@ PostSchema.statics = {
     if (type === 'top') {
       sort = { score: -1 };
     }
-
-    // return this.find().count()
-    //   .then((count) => {
-    //     numberOfPages = Math.floor(count / limit);
-    //
-    //     return this.find(query)
-    //       .sort(sort)
-    //       .limit(limit)
-    //       .exec();
-    //   })
-
-    return this.find(query, 'content title date mp3 link score featuredImage upvoted downvoted tags categories')
+    const queryPromise = this.find(query, 'content title date mp3 link score featuredImage upvoted downvoted tags categories')
       .sort(sort)
-      .limit(limitOption)
-      .then((postsFound) => {
+      .limit(limitOption);
+
+    if (!user) {
+      return queryPromise.then((postsFound) => {
         posts = postsFound;
         // Flip direct back
         if (dateDirection === 1) {
           posts.reverse();
         }
-        if (!user) return posts;
+        return postsFound;
+      });
+    }
 
+    return queryPromise
+      .populate({
+        path: 'favoritedByUser',
+        select: 'active',
+        match: { userId: user._id }
+      }).exec()
+      .then((postsFound) => {
+        posts = postsFound;
+        // Flip direct back
```

Comment:
I know this is probably copied from above but is it just supposed to say: Flip direction back?

Reply:
You're right I'm going to update this. And also not repeat myself by conditionally adding another chained method...

But I wan't to say flip direction back and why. But why? This use case would be if the client uses `createdAfter=2016-04-01` for example. With a limit of 10 this would be the first ten after 2016-04-01 by date ascending. Then when returning the result we flip the order so it's these ten but with the most recent among them first. It's not used in the web anyways, but my guess is if a client were to use this query you would want the oldest first, no?